apps/envited-x-data-space/common/validator/shacl/shacl.ts

```diff
@@ -38,6 +41,15 @@ export const _validateShaclFile =
   }) =>
   async (file: File) => {
     try {
+      const assetExists = await checkIfAssetExists(file)
```

Comment:
Here I might need some more info but ideal situation:

The shacl validator does nothing else but take a shacl file and validate a provided file against it. Nothing more nothing less.

By first checking if an asset exists you now directly couple our concept of what was an abstract provided file into a concrete concept of an asset as we see it *and* in one swoop introduce an extra external dependency. 

That's why these functions should be split up and first check if the assets exists, and only then move over to anything that has to do with the provided shacls